File: Backend/dependency-dotnet/MHP.CodingChallenge.Backend.Dependency.Notifications/EmailHandler.cs
using System;
using MHP.CodingChallenge.Backend.Dependency.Inquiry;
using Microsoft.Extensions.Logging;

namespace MHP.CodingChallenge.Backend.Dependency.Notifications
{
    public class EmailHandler
    {
        public virtual void SendEmail(Inquiry.Inquiry inquiry)
        {
            Console.WriteLine(string.Format("sending email for: {0}", inquiry.ToString()));
        }
    }
}

File: Backend/dependency-kotlin/notifications/src/main/kotlin/com/mhp/coding/challenges/dependency/notifications/EmailHandler.kt
package com.mhp.coding.challenges.dependency.notifications

import com.mhp.coding.challenges.dependency.inquiry.Inquiry
import com.mhp.coding.challenges.dependency.notifications.EmailHandler
import mu.KotlinLogging
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

private val logger = KotlinLogging.logger {}

@Component
class EmailHandler {
    fun sendEmail(inquiry: Inquiry) {
        logger.info {
            "Sending email for: $inquiry"
        }
    }
}
